backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -87,9 +105,38 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
                 return;
             }
 
+            if (!await LocalizationService.RobotIsLocalized(robot.Id))
+            {
+                try { await LocalizationService.EnsureRobotWasCorrectlyLocalizedInPreviousMissionRun(robot.Id); }
+                catch (Exception ex) when (ex is LocalizationFailedException or RobotNotFoundException or MissionNotFoundException or MissionException or TimeoutException)
+                {
+                    //TODO Handle failed localization - Cancel all the missions?
+                    _logger.LogError("Could not confirm that the robot was correctly localized and the scheduled missions for the deck will be cancelled");
+                }
+            }
+
             if (MissionScheduling.MissionRunQueueIsEmpty(await MissionService.ReadMissionRunQueue(robot.Id)))
             {
                 _logger.LogInformation("The robot was changed to available but there are no mission runs in the queue to be scheduled");
+
+                var lastExecutedMissionRun = await MissionService.ReadLastExecutedMissionRunByRobot(robot.Id);
+                if (lastExecutedMissionRun is null)
+                {
+                    _logger.LogError("Could not find last executed mission run for robot");
+                    return;
+                }
+
+                if (!lastExecutedMissionRun.IsDriveToMission())
+                {
+                    try { await ReturnToHomeService.ScheduleReturnToHomeMissionRun(robot.Id); }
+                    catch (Exception ex) when (ex is RobotNotFoundException or AreaNotFoundException or DeckNotFoundException or PoseNotFoundException)
+                    {
+                        //TODO Handle failed return to home - Try again for a number of times?
```

Comment:
Ensure to break the localization if this fails

Reply:
`CurrentArea` is set to `null`, to handle in another PR the error handling behaviour of notifying the frontend